src/lib/components/Legend/Legend.js

```diff
@@ -265,17 +265,29 @@ export class Legend extends React.Component {
           );
         }
         if (fillLayerNoBreaks && !layer.parent) {
+          const hasShape = (layer.categories && layer.categories.shape && layer.categories.shape.length);
           const fillWidth = (
             100 / layer.categories.color.filter(c => c !== 'transparent').length
           ).toString();
-
+            
           layer.categories.color.forEach((color, index) => {
             if (color !== 'transparent') {
-              background.push(
-                <li key={index} style={{ background: color, width: `${fillWidth}%` }}>
-                  {layer.categories.label[index]}
-                </li>
-              );
+              if (hasShape) {
```

Comment:
Should this check for `layer.categories.shape[index]` as well? Or are we fine with the `<li>` and `<span>` having class: `undefined`? Can we give the those classes a backup if `layer.categories.shape[index]` doesn't exist?